kalpsdk/read_transaction.go

```diff
@@ -6,11 +6,77 @@ import (
 	"fmt"
 	"strconv"
 	"strings"
+	"time"
 
 	//Third party Libs
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
+// IsSmartContractOwner checks if the caller is the owner of the smart contract
+// by comparing the userID of the caller with the stored smart contract owner ID.
+//
+// Returns:
+//   - bool: A boolean value indicating whether the caller is the owner of the smart contract.
+//   - error: An error if the operation fails.
+func (ctx *TransactionContext) IsSmartContractOwner() (bool, error) {
+	userID, err := ctx.GetUserID()
+	if err != nil {
+		return false, err
+	}
+
+	ownerId, err := ctx.GetStub().GetState("smartContractOwner")
```

Comment:
I would define this constant value as a const variable and then use it here. it would be helpful in future to modify or add new roles.

Reply:
done